client/src/components/Chat/Messages/Content/ToolCallInfo.tsx

```diff
@@ -57,16 +57,21 @@ export default function ToolCallInfo({
   // Extract ui_resources from the output to display them in the UI
   let uiResources: UIResource[] = [];
   if (output?.includes('ui_resources')) {
-    const parsedOutput = JSON.parse(output);
-    const uiResourcesItem = parsedOutput.find(
-      (contentItem) => contentItem.metadata === 'ui_resources',
-    );
-    if (uiResourcesItem?.text) {
-      uiResources = JSON.parse(atob(uiResourcesItem.text)) as UIResource[];
+    try {
+      const parsedOutput = JSON.parse(output);
```

Comment:
@sbruel you said [here](https://github.com/danny-avila/LibreChat/pull/9299#discussion_r2310684499):

> we probably need more solid error handling if the output includes ui_resources as text, for some reason, without being a proper ui_resource

This test case shows that this is properly handled:

https://github.com/danny-avila/LibreChat/blob/d59784c660bb6140863852b10c8ecaf82607aa23/client/src/components/Chat/Messages/Content/__tests__/ToolCallInfo.test.tsx#L169-L189